### PR TITLE
Default parameter error surface blending service 11

### PR DESCRIPTION
**Pull request by pdaelm**
Fix to get path planning parameters from ros param server if parameters defined in GUI are not coherent, and `~/.ros/godel_*.msg` files are missing.

### CODE REVIEW

**Comment by miguelprada** [comment]
I can confirm this works in my limited trials.

It would be even better if this populated the option menus in the GUI with the defaults read from the parameter server as well. Do you think this is feasible or would it complicate things too much?

**Comment by pdaelm** [comment]
I can try to do it, I let you know

File: godel_surface_detection/src/services/surface_blending_service.cpp [hunk 1]
```diff
@@ -247,11 +247,20 @@ void SurfaceBlendingService::save_blend_parameters(const std::string& filename)
...
 {
+  using godel_param_helpers::loadParam;
+
   if(godel_param_helpers::fromFile(filename, path_planning_params_))
   {
     return true;
   }
-  return false;
+  // otherwise default to the parameter server
+  ros::NodeHandle nh ("path_planning_params");
```
Comment (miguelprada) [inline godel_surface_detection/src/services/surface_blending_service.cpp:257]: Other parameter _sets_ are read from specific namespaces (e.g. [`"~/blend_plan"`](https://github.com/tecnalia-advancedmanufacturing-robotics/godel/blob/94a63a2ad693f3d7941fe35584b481da4080ae47/godel_surface_detection/src/services/surface_blending_service.cpp#L225)). Shouldn't this be done similarly?

File: godel_surface_detection/src/services/surface_blending_service.cpp [hunk 2]
```diff
@@ -247,11 +247,20 @@ void SurfaceBlendingService::save_blend_parameters(const std::string& filename)
...
+  return  loadParam(nh, "discretization", path_planning_params_.discretization) &&
+          loadParam(nh, "margin", path_planning_params_.margin) &&
+          loadParam(nh, "overlap", path_planning_params_.overlap) &&
+          loadParam(nh, "safe_traverse_height", path_planning_params_.traverse_height) &&
+          loadParam(nh, "scan_width", path_planning_params_.scan_width) &&
+          loadParam(nh, "tool_radius", path_planning_params_.tool_radius);
```
Comment (miguelprada) [inline godel_surface_detection/src/services/surface_blending_service.cpp:263]: If any of the `loadParam` call fails the [remaining parameters will not be loaded](https://en.wikipedia.org/wiki/Short-circuit_evaluation), yet [only a warning is raised](https://github.com/tecnalia-advancedmanufacturing-robotics/godel/blob/94a63a2ad693f3d7941fe35584b481da4080ae47/godel_surface_detection/src/services/surface_blending_service.cpp#L94-L95) and initialization continues. This doesn't make a lot of sense to me.

At the very least I think we should always try to read all parameters.

**Review by miguelprada — CHANGES_REQUESTED** [review]
Other than the inline comments, I tested it and it seems to work as expected.

**Comment by pdaelm** [comment]
Ping

**Pull request merged**